src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -1709,6 +1716,19 @@ AssignTasksToConnectionsOrWorkerPool(DistributedExecution *execution)
 												sizeof(TaskPlacementExecution *));
 		shardCommandExecution->placementExecutionCount = placementExecutionCount;
 
+		TupleDestination *tupleDest = task->tupleDest;
+		uint32 numQueries = tupleDest ? tupleDest->numQueries(tupleDest) : 0;
+		shardCommandExecution->attributeInputMetadata = palloc0(numQueries *
+																sizeof(AttInMetadata *));
+
+		for (uint32 queryIndex = 0; queryIndex < numQueries; queryIndex++)
+		{
+			TupleDesc tupleDescriptor = tupleDest->tupleDescForQuery(tupleDest,
+																	 queryIndex);
+			shardCommandExecution->attributeInputMetadata[queryIndex] =
```

Comment:
Could tupleDescForQuery return NULL? (e.g. for DML)

Reply:
In addition, should there be a /dev/null-style tuple destination for DML without returning/DDL?